ansible/playbooks/roles/repository/files/download-requirements/src/mode/red_hat_family_mode.py

```diff
@@ -106,85 +100,94 @@ def _add_third_party_repositories(self):
 
-        unique_collected_prereqs: Set = set(collected_prereqs)
-        for prereq in unique_collected_prereqs:
-            self._tools.yumdownloader.download_packages([prereq],
-                                                        arch=self._cfg.os_arch.value,
-                                                        exclude='*i686',
-                                                        destdir=prereqs_dir)
-            logging.info(f'- {prereq}')
+        logging.info('- prereq-packages to download: %s', collected_prereqs)
 
-        return unique_collected_prereqs
+        self._tools.dnf_download.download_packages(collected_prereqs,
+                                                   archlist=self.__archs,
+                                                   exclude='*i686',
+                                                   destdir=prereqs_dir)
+        return collected_prereqs
 
     def _download_packages(self):
-        downloaded_prereqs: Set = self.__download_prereq_packages()
-
-        packages: List[str] = self._requirements['packages']['from_repo']
-        packages_to_download: List[str] = []
-        for package in packages:
-            # package itself
-            package_name = self._tools.repoquery.query(package,
-                                                       queryformat='%{ui_nevra}',
-                                                       arch=self._cfg.os_arch.value)[0]
-
-            packages_to_download.append(package_name)
-
-            # dependencies
-            packages_to_download.extend(self._tools.repoquery.get_dependencies(package,
-                                                                               queryformat='%{name}.%{arch}',
-                                                                               arch=self._cfg.os_arch.value))
-
-        for package in set(packages_to_download):
-            if package not in downloaded_prereqs:
-                logging.info(f'- {package}')
-                self._tools.yumdownloader.download_packages([package],
-                                                            arch=self._cfg.os_arch.value,
-                                                            exclude='*i686',
-                                                            destdir=self._cfg.dest_packages)
+        downloaded_prereq_packages: List[str] = self.__download_prereq_packages()
+
+        packages: List[str] = sorted(set(self._requirements['packages']['from_repo']))
+
+        # packages
+        queried_packages: List[str] = self._tools.repoquery.query(packages,
+                                                                  queryformat='%{name}-%{version}-%{release}.%{arch}',
+                                                                  archlist=self.__archs)
+
+        packages_to_download: List[str] = sorted(set(queried_packages) - set(downloaded_prereq_packages))
+
+        logging.info('- packages to download: %s', packages_to_download)
```

Comment:
We decided to use f-strings with logging, please apply that to every logging call.

Reply:
Changed to f-strings.

Unfortunately there is no option to force f-strings for logging using Pylint (`fstr` option was added in [2.4](http://pylint.pycqa.org/en/latest/whatsnew/2.4.html) and removed in [2.5](http://pylint.pycqa.org/en/latest/whatsnew/2.5.html)). 

The only thing we can do is to disable the default Pylint's rule with `disable=logging-fstring-interpolation` and I did it.

Since we're not able to ensure `fstr` for logging by linter I would opt for allowing both styles.

For more info, see [related issue and comment](https://github.com/PyCQA/pylint/issues/1788#issuecomment-410381475).